addons/l10n_in/models/account_invoice.py

```diff
@@ -71,23 +71,49 @@ def _compute_l10n_in_state_id(self):
 
     @api.depends('l10n_in_state_id')
     def _compute_fiscal_position_id(self):
-        other_country = None
-        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_sale_document(include_receipts=True))
-        for move in in_moves:
-            state_id = move.l10n_in_state_id.id
-            country_id = move.l10n_in_state_id.country_id.id
-            if move.l10n_in_state_id.l10n_in_tin == '96':  # Foreign Country
-                other_country = other_country or self.env['res.country'].search([('code', '!=', 'IN')], limit=1)
-                country_id = other_country.id
-                state_id = False
-            virtual_partner = self.env['res.partner'].new({
-                'state_id': state_id,
-                'country_id': country_id,
-            })
-            move.fiscal_position_id = self.env['account.fiscal.position'].with_company(
-                move.company_id
-            )._get_fiscal_position(virtual_partner)
-        super(AccountMove, self - in_moves)._compute_fiscal_position_id()
+        computed_ids = []
+        for move in self:
+            if (
+                move.country_code != 'IN'
+                or not move.is_invoice(include_receipts=True)
+                or move.partner_shipping_id.property_account_position_id
+                or move.partner_id.property_account_position_id
+            ):
+                continue
+            dummy_state_id = (
+                move.l10n_in_gst_treatment == 'special_economic_zone'
+                and self.env.ref('l10n_in.state_in_oc')
+                or False
+            )
+            # If move type is sale, compare place of supply with company state
+            if not dummy_state_id and move.is_sale_document(include_receipts=True):
+                dummy_state_id = move.l10n_in_state_id
+            # If move type is purchase, compare place of supply with vendor state
+            elif (
+                not dummy_state_id
+                and move.is_purchase_document(include_receipts=True)
+                and move.partner_id.country_id.code == 'IN'
+            ):
```

Comment:
No it's not ruled out, here we are checking partner's country code not move's country code, if only do it for purchase when the partner's country code is `IN`. In case of overseas partner the super should be called.

Reply:
@hamo-odoo So in the case of place of supply OC or country not India, it will just apply the import through the super.